### PR TITLE
[PCFG-225] Enabled ApiConfig for Production

### PR DESCRIPTION
**Pull request by jacopocarlini**
### List of changes

- Added auto-scaling to ApiConfig
- removed variable api_config_enabled -> apiconfig enabled in all environments

### Motivation and context

To deploy in production

### Type of changes

- [ ] Add new resources
- [x] Update configuration to existing resources
- [ ] Remove existing resources

### Does this introduce a change to production resources with possible user impact?

- [ ] Yes, users may be impacted applying this change
- [x] No

### Does this introduce an unwanted change on infrastructure? Check terraform plan execution result

- [ ] Yes
- [x] No

### Other information

---

### CODE REVIEW

File: src/api_config.tf [hunk 1]
```diff
@@ -139,3 +139,67 @@ resource "azurerm_monitor_scheduled_query_rules_alert" "apiconfig_db_healthcheck
...
+        time_aggregation         = "Average"
+        operator                 = "GreaterThan"
+        threshold                = 250
+        divide_by_instance_count = false
+      }
+
+      scale_action {
+        direction = "Increase"
+        type      = "ChangeCount"
+        value     = "2"
```
Comment (pasqualespica) [inline src/api_config.tf:176]: replace with `1`

File: src/env/prod/terraform.tfvars [hunk 2]
```diff
@@ -118,7 +118,7 @@ mock_ec_enabled  = false
 mock_psp_enabled = false
 
 # api_config
-api_config_enabled = false
+api_config_enabled = true
```
Comment (pasqualespica) [inline src/env/prod/terraform.tfvars:121]: now that api-config is enabled in all environments is `api_config_enabled` variable still needed?

**Review by pasqualespica — CHANGES_REQUESTED** [review]

File: src/env/prod/terraform.tfvars (hunk 2 above)
Comment (jacopocarlini) [reply src/env/prod/terraform.tfvars]: variable removed

**Review by pasqualespica — APPROVED** [review]
great 🚀

**Pull request merged**